Generic Bash Wrapper around Terragrunt Operations

anocendi opened the pull request:
First candidate implementation of bash wrapper around terragrunt
operations.

The configuration to be applied can be supplied via json format
configuration file, which defaults to the configs directory at the same
level as the wrapper script.

Signed-off-by: Soe San Win <soe.san.win.oss@gmail.com>

Commit:
Generic Bash Wrapper around Terragrunt Operations

First candidate implementation of bash wrapper around terragrunt
operations.

The configuration to be applied can be supplied via json format
configuration file, which defaults to the configs directory at the same
level as the wrapper script.

Signed-off-by: Soe San Win <soe.san.win.oss@gmail.com>

robc-io commented:
A big part of this was so we can have one list of directories and then loop down and up (apply and destroy) and don't need to worry about maintaining two lists. Can you build the destroy loop like I have it reversing the apply?

anocendi commented:
So, just one list, apply will go one direction, and destroy will go in the reverse? Cool, will do that.

Commit:
Generic Bash Wrapper amend #1

Addressed Rob's comments and concerns on the Generic Bash Wrapper.

Instead of maintaining two lists in the json configuration, keep only
one for apply, and destroy will call the steps in reverse direction.

Signed-off-by: Soe San Win <soe.san.win.oss@gmail.com>
